src/QuartzNET-DynamoDB/DataModel/Storage/Repository.cs

```
@@ -4,16 +4,26 @@
 using System.Net;
 using Amazon.DynamoDBv2;
 using Amazon.DynamoDBv2.Model;
+using Polly;
 
 namespace Quartz.DynamoDB.DataModel.Storage
 {
-    public class Repository<T> : IRepository<T>, IDisposable where T : IInitialisableFromDynamoRecord, IConvertibleToDynamoRecord, IDynamoTableType, new()
+    public class Repository<T> : IRepository<T> where T : IInitialisableFromDynamoRecord, IConvertibleToDynamoRecord, IDynamoTableType, new()
     {
-        private AmazonDynamoDBClient _client;
+        private readonly AmazonDynamoDBClient _client;
+        private readonly Policy _writeRetryPolicy;
+        private readonly Policy _readRetryPolicy;
 
         public Repository(AmazonDynamoDBClient client)
         {
             _client = client;
+            _writeRetryPolicy = Policy
+                .Handle<ProvisionedThroughputExceededException>()
+                .WaitAndRetry(5, ExponentialBackoffWithRandomVariation.CalculateWaitDuration);
+
+            _readRetryPolicy = Policy
+             .Handle<ProvisionedThroughputExceededException>()
+             .WaitAndRetry(3, ExponentialBackoffWithRandomVariation.CalculateWaitDuration);
```

out of curiosity, how come you chose to go for 3 retries on reads but 5 for writes?

---

Some logic thinking that reads should be faster, so want less retries. I was pretty tired by that point. Think i should make them the same?

---

faster in what sense? sorry, not sure i follow 😕

---

Faster as in, fails faster due to less retries. Clearly this is not overly rational. I think I'll refactor the class to have just one retry policy...

---

I was thinking that if you call an operation that say, just returns jobs and it takes 5 minutes - that's kind of annoying and should fail faster?

---

see what you mean :)

---

Should we make an issue to make read and write policies the same? Or just leave it?

---

i'm happy with it as is, can look into it if this ever appears to be problematic.

---

:thumbsup: